### PR TITLE
get rid of deprecation warnings from `RecoVertex` subsystem

### PR DESCRIPTION
**Pull request by mmusich**
#### PR description:

Trivial PR, title says it all.
Part of the migration in #36404.
Went systematically through all of the `CMSDEPRECATED_X` warnings in the `RecoVertex` subsystem from `CMSSW_12_3_CMSDEPRECATED_X_2022-01-21-2300` and removed deprecated API calls.

#### PR validation:

`cmssw` compiles

#### if this PR is a backport please specify the original PR and why you need to backport that PR:

N/A

### CODE REVIEW

**Comment by mmusich** [comment]
@cmsbuild, please test

File: RecoVertex/GaussianSumVertexFit/test/GsfTest.cc [hunk 1]
```diff
@@ -36,19 +89,14 @@ GsfTest::GsfTest(const edm::ParameterSet& iConfig) : theConfig(iConfig) {
...
 void GsfTest::analyze(const edm::Event& iEvent, const edm::EventSetup& iSetup) {
   if (not tree) {
-    edm::ESHandle<MagneticField> magField;
-    iSetup.get<IdealMagneticFieldRecord>().get(magField);
-    tree.reset(new SimpleVertexTree("VertexFitter", magField.product()));
+    const MagneticField* magField = &iSetup.getData(estoken_mf);
+    tree.reset(new SimpleVertexTree("VertexFitter", magField));
   }
 
   try {
```
Comment (clacaputo) [inline RecoVertex/GaussianSumVertexFit/test/GsfTest.cc:102]: Could you please move to `cms::Exception`? Also, there are a few `cout` that can be changed to `edm::LogPrint`

File: RecoVertex/GaussianSumVertexFit/test/TrackMix.cc [hunk 2]
```diff
@@ -1,37 +1,65 @@
...
 
-TrackMix::~TrackMix() {}
-
-void TrackMix::beginJob() {}
-
-void TrackMix::endJob() {}
+TrackMix::~TrackMix() = default;
 
 void TrackMix::analyze(const edm::Event& iEvent, const edm::EventSetup& iSetup) {
   try {
```
Comment (clacaputo) [inline RecoVertex/GaussianSumVertexFit/test/TrackMix.cc:65]: Could you please move to `cms::Exception`? Also, there are a few `cout` that can be changed to `edm::LogPrint`

**Review by clacaputo — COMMENTED** [review]
Could you please move to `cms::Exception` in the following `EDAnalyzer`? Also, there are a few `cout` that can be changed to `edm::LogPrint`

- `GsfTest`
- `TrackMix`
- `KineExample`

File: RecoVertex/GaussianSumVertexFit/test/TrackMix.cc (hunk 2 above)
Comment (mmusich) [reply RecoVertex/GaussianSumVertexFit/test/TrackMix.cc:65]: done in the last push.

File: RecoVertex/GaussianSumVertexFit/test/GsfTest.cc (hunk 1 above)
Comment (mmusich) [reply RecoVertex/GaussianSumVertexFit/test/GsfTest.cc:102]: Done in the last push.

**Comment by mmusich** [comment]
@cmsbuild, please test

**Comment by clacaputo** [comment]
+reconstruction

- technical
- reco change related to https://github.com/cms-sw/cmssw/issues/36552

**Comment by perrotta** [comment]
+1

**Pull request merged**